main.py

```diff
@@ -356,6 +362,53 @@ def __init__(self, *args):
 		self.actionExit.triggered.connect(self.onExitClicked)
 		self.actionSettings.triggered.connect(self.onSettingsClicked)
 
+		if not "search_for_updates" in self.settings.settings or not "buildId" in self.settings.settings:
+			self.settings.settings["search_for_updates"] = False
+			self.settings.settings["buildId"] = ""
+			self.settings.save()
+
+		if "search_for_updates" in self.settings.settings:
+			if self.settings.settings["search_for_updates"] == True:
+				if os.name == "nt":
+					projectJson = json.loads(requests.get("https://ci.appveyor.com/api/projects/mborgerson/xqemu-c5j6o/branch/master").text)
+					jobId = projectJson["build"]["jobs"][0]["jobId"]
+					print("Job ID: " + jobId)
+
+					if not jobId == self.settings.settings["buildId"]:
+						reply = QMessageBox.question(self, 'Update', "A new update is available, would you like to update?", QMessageBox.Yes, QMessageBox.No)
+						if reply == QMessageBox.Yes:
+							dir = os.path.dirname(os.path.abspath(self.settings.settings["xqemu_path"]))
+							print(dir)
+							if os.path.isdir(dir):
+								buildjobsJson = json.loads(requests.get("https://ci.appveyor.com/api/buildjobs/" + jobId + "/artifacts").text)
+
+								fileName = buildjobsJson[0]["fileName"]
+								print("Filename: " + fileName)
+
+								binaryUrl = "https://ci.appveyor.com/api/buildjobs/" + jobId + "/artifacts/" + fileName
+								print("Downloading " + fileName + " from " + binaryUrl)
+
+								with open(fileName, "wb") as file:
+									response = requests.get(binaryUrl)
+									file.write(response.content)
+								print("Downloaded")
+
+								self.settings.settings["buildId"] = jobId
+
+								print("Extracting...")
+								with zipfile.ZipFile(fileName, "r") as z:
+									z.extractall(dir)
+
+								os.remove(fileName)
```

Comment:
If we don't want to keep files, we should be using `StringIO` or something (which differs for Python 2 and Python 3).

However, I'd argue that we want to *keep* files, to avoid downloading builds the user already has (in case of regressions this will be handy).